control-plane-operator/controllers/hostedcontrolplane/etcd/reconcile.go

```diff
@@ -96,6 +103,19 @@ func ReconcileStatefulSet(ss *appsv1.StatefulSet, p *EtcdParams) error {
 		util.BuildContainer(etcdHealthzContainer(), buildEtcdHealthzContainer(p, ss.Namespace)),
 	}
 
+	// only deploy etcd-defrag-controller container in HA mode
```

Comment:
>// only deploy etcd-defrag-controller container in HA mode

why?

Reply:
defragmentation will take the etcd instance offline, so if we do defragmentation in non-HA mode (one etcd instance), the whole cluster would go offline until defrag is done.

Reply:
Yes and this mirrors what is happening in OCP.

Reply:
I see, thanks! can we add that to the comment?